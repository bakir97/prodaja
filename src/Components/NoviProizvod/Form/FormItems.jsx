import React from "react";
import CircularProgress from "@material-ui/core/CircularProgress";
import SelectField from "./SelectField";
import TextAreaComponent from "../../../Layout/FormElements/TextArea";
import { Grid, Button } from "@material-ui/core";
import FormInput from "./FormInput";
import { Field } from "redux-form";
import style from "./style";

export default ({ loading, oldPictures, pictures }) => {
  return (
    <>
      <Grid item xs={12} container justify="center">
        <FormInput type="text" name="naslov" label="Naslov" />
        <Field type="text" name="opis" component={TextAreaComponent} />
        <SelectField />
        <Grid item xs={6} style={{ paddingRight: 5 }}>
          <FormInput type="text" name="tezina" label="Tezina" />
        </Grid>
        <Grid style={{ paddingLeft: 5 }} item xs={6}>
          <FormInput type="text" name="cijena" label="Cijena" />
        </Grid>
      </Grid>
      <Grid item xs={12} style={{ textAlign: "center" }}>
        {loading ? (
          <CircularProgress />
        ) : (
          <Button
            disabled={oldPictures.length + pictures.length === 0}
            type="submit"
            style={style.button}
          >
            Objavi Proizvod
          </Button>
        )}
      </Grid>
    </>
  );
};
