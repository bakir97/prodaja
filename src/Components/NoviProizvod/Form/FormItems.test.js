import FormItems from "./FormItems";
describe("FormItems", () => {
  const wrapper = shallow(<FormItems oldPictures={[]} pictures={[]} />);
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
