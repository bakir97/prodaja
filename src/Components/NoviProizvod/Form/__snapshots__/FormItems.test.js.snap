// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItems should render 1`] = `
<Fragment>
  <WithStyles(Grid)
    container={true}
    item={true}
    justify="center"
    xs={12}
  >
    <_default
      label="Naslov"
      name="naslov"
      type="text"
    />
    <Field
      component={[Function]}
      name="opis"
      type="text"
    />
    <_default />
    <WithStyles(Grid)
      item={true}
      style={
        Object {
          "paddingRight": 5,
        }
      }
      xs={6}
    >
      <_default
        label="Tezina"
        name="tezina"
        type="text"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      style={
        Object {
          "paddingLeft": 5,
        }
      }
      xs={6}
    >
      <_default
        label="Cijena"
        name="cijena"
        type="text"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    style={
      Object {
        "textAlign": "center",
      }
    }
    xs={12}
  >
    <WithStyles(Button)
      disabled={true}
      style={
        Object {
          "fontFamily": "Kalam",
          "fontSize": 20,
          "marginTop": 20,
          "paddingBottom": 3,
        }
      }
      type="submit"
    >
      Objavi Proizvod
    </WithStyles(Button)>
  </WithStyles(Grid)>
</Fragment>
`;
